hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerRegistration.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.spi.impl.listener;
+
+import com.hazelcast.client.impl.spi.EventHandler;
+import com.hazelcast.client.impl.spi.impl.ListenerMessageCodec;
+import com.hazelcast.internal.nio.Connection;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ClientListenerRegistration {
+
+    private final ListenerMessageCodec codec;
+    private final EventHandler handler;
+    private final Map<Connection, ClientConnectionRegistration> registrations = new ConcurrentHashMap<>();
+
+    ClientListenerRegistration(EventHandler handler, ListenerMessageCodec codec) {
+        this.handler = handler;
+        this.codec = codec;
+    }
+
+    public ListenerMessageCodec getCodec() {
+        return codec;
+    }
+
+    public EventHandler getHandler() {
+        return handler;
+    }
+
+    Map<Connection, ClientConnectionRegistration> getConnectionRegistrations() {
```

Comment:
exposing a private variable to outside this class (even though the same package).

Alternative approach: Do not expose it outside and provide put, get, iterator() methods directly on `ClientListenerRegistration` class.

Reply:
One can argue same thing about other two fields also. I would like to think that `ClientListenerRegistration` is just a holder class and we do not do any operations `put` `get` etc.. on it.